### ISSUE TITLE
transformer is called way to often which makes large rete networks unusable slowly

### ISSUE THREAD

**Issue by pokecheater** [retejs/rete#586]
Hey Retejs Team,

thx for your great work :).

I stumbled recently over an problem with the rete js designer. I have a network of retejs nodes which is has 207 connections and the most of them go from one to the next node directly. I have a node which summarize all those nodes but has the possibility to unfold it (which basicly extract the inner network into the outer). For this I create nodes and connect them. All these steps happens in a reasobable time but afterwards during the render process everything gets stucked for a long while. When the ui comes back to life any tiny change on the connections or node positions retrigger this behaviour.

I was able to identify that my connection path plugin transformer is called way to often (in my opinion).
I have written a console log inside this function ( see screen shot 1) and now my js debugger tells me that it is called >38.000 times (screen shot 2) (note that this count still increases while i write this xD). A 

So I tried to set up a simple network with 2 nodes and one connection. And even here on each change this method is called 20 times (screenshot 3). (please note that I 

Screenshot 1

![Bildschirmfoto 2022-09-14 um 14 01 39](https://user-images.githubusercontent.com/20928617/190150331-89bf6cc4-a5a8-4fcf-8e8b-3e2a5632144b.png)

Screenshot 2
  
<img width="369" alt="Bildschirmfoto 2022-09-14 um 14 06 07" src="https://user-images.githubusercontent.com/20928617/190150362-bbcda628-b550-4c10-a750-8194231ca6f0.png">

Screenshot 3
<img width="917" alt="Bildschirmfoto 2022-09-14 um 14 11 22" src="https://user-images.githubusercontent.com/20928617/190150399-3eca1a87-78f9-4c12-8472-85ee1e9c7668.png">

Can you help me out with this issue? Do you have any information on that?

Greetz

**Comment by barnesew** [comment]
In case it helps, I believe this might be related to v1.4.9. I also utilize rete in an application with hundreds of connections and noticed a significant performance impact when we upgraded from v1.4.8 to v1.4.9. I believe something about the deferred connection rendering that was recently introduced is causing the "Animation Frame Fired" event to be called repeatedly. Below is the call tree after I selected one of the nodes on the screen (v1.4.9).

![Screen Shot 2022-10-04 at 11 26 33 AM](https://user-images.githubusercontent.com/18120656/193860952-b1673de3-0b5f-4516-a796-d1212afc1c97.png)
